studio/src/build_manager/build_server.rs

```diff
@@ -281,15 +236,21 @@ impl BuildConnection {
                 "--message-format=json".into(),
             ]
         };
-        
+
         let http = format!("{}/{}", http, cmd_id.0);
-        let env = [
+        let mut env = vec![
             ("MAKEPAD_STUDIO_HTTP", http.as_str()),
             ("MAKEPAD", "lines")
         ];
 
-        let process = ChildProcess::start("rustup", &args, path, &env).expect("Cannot start process");
-        
+        // Default to nightly rustc but don't overwrite any user request for a
+        // specific nightly version.
+        if !env::var("RUSTUP_TOOLCHAIN").map_or(false, |toolchain| toolchain.contains("nightly")) {
+            env.push(("RUSTUP_TOOLCHAIN", "nightly"));
```

Comment:
The `RUSTUP_TOOLCHAIN` env var is read by rustup to determine which toolchain to use when the `+toolchain` cli arg or `rustup run` isn't used and set for any command it invokes with the toolchain that is actually used.